src/hooks/use-handle-snapshot.ts

```diff
@@ -9,43 +9,139 @@
  *
  * All portions of the code written by Atypon Systems LLC are Copyright (c) 2019 Atypon Systems LLC. All Rights Reserved.
  */
+import { Model } from '@manuscripts/json-schema'
 import { usePermissions } from '@manuscripts/style-guide'
 import { trackCommands } from '@manuscripts/track-changes-plugin'
+import { ContainedModel, encode, ManuscriptNode } from '@manuscripts/transform'
+import isEqual from 'lodash-es/isEqual'
+import { EditorView } from 'prosemirror-view'
 
-import { useEditorStore } from '../components/track-changes/useEditorStore'
+import { post } from '../quarterback/api/methods'
 import { getDocWithoutTrackContent } from '../quarterback/getDocWithoutTrackContent'
-import { usePouchStore } from '../quarterback/usePouchStore'
-import { useSnapshotStore } from '../quarterback/useSnapshotStore'
+import { ISaveSnapshotResponse } from '../quarterback/types'
 import { useStore } from '../store'
 
-export const useHandleSnapshot = (storeExists = true) => {
-  const { execCmd, docToJSON } = useEditorStore()
-  const { saveSnapshot } = useSnapshotStore()
+export type Ok<T> = {
+  data: T
+}
+export type Error = {
+  err: string
+  code: number
+}
+export type Maybe<T> = Ok<T> | Error
+
+const EXCLUDED_KEYS = [
+  'id',
+  '_id',
+  '_rev',
+  '_revisions',
+  'sessionID',
+  'createdAt',
+  'updatedAt',
+  'owners',
+  'manuscriptID',
+  'containerID',
+  'src',
+  'minWordCountRequirement',
+  'maxWordCountRequirement',
+  'minCharacterCountRequirement',
+  'maxCharacterCountRequirement',
+] as (keyof Model)[]
+
+const hasChanged = (a: Model, b: Model): boolean => {
+  return !!Object.keys(a).find((key: keyof Model) => {
+    if (EXCLUDED_KEYS.includes(key)) {
+      return false
+    }
+    return !isEqual(a[key], b[key])
+  })
+}
+
+const saveDoc = async (
```

Comment:
this function and all what it used for it (hasChanged function, EXCLUDED_KEYS, type Maybe, type Ok, type Error) is too much info for this file. useHandleSnapshot should only import saveDoc. It can live in the store if it needs to be created dynamically of move into a different api related file / folder.

Reply:
moved to `savingUtilities.ts`